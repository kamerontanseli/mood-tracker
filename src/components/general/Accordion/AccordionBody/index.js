import "./AccordionBody.css";
import React from "react";

const AccordionBody = ({ children }) => (
  <div className="AccordionBody">{children}</div>
);

export default AccordionBody;
