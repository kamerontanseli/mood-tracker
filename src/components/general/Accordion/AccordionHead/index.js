import "./AccordionHead.css";
import React from "react";

const AccordionHead = ({ children }) => (
  <div className="AccordionHead">{children}</div>
);

export default AccordionHead;
