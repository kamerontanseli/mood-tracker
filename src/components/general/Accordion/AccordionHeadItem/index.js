import "./AccordionHeadItem.css";
import React from "react";

const AccordionHeadItem = ({ children }) => (
  <div className="AccordionHeadItem">{children}</div>
);

export default AccordionHeadItem;
