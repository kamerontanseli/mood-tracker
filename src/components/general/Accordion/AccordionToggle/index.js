import React from "react";
import Button from '../../Button'

const AccordionToggle = ({ toggled, ...props }) => (
  <Button {...props}>
    {toggled ? <span>&uarr;</span> : <span>&darr;</span>}
  </Button>
);

export default AccordionToggle;
