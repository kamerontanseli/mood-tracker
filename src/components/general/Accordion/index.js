import React from "react";
import Panel from "../../general/Panel";

export default class Accordion extends React.Component {
  state = {
    toggled: false
  };

  toggle = () =>
    this.setState(state => ({
      toggled: !state.toggled
    }));

  render() {
    return (
      <Panel>
        {this.props.children({
          onToggle: this.toggle,
          toggled: this.state.toggled
        })}
      </Panel>
    );
  }
}
