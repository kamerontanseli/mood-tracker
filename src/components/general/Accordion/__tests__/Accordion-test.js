import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Accordion from '..'

describe('toggle', () => {
  it('should toggle state.toggle', () => {
    const spy = jest.fn()
    const wrapper = shallow(<Accordion>{spy}</Accordion>)
    expect(spy).toHaveBeenCalled()
    wrapper.instance().toggle()
    wrapper.update()
    expect(wrapper.state('toggled')).toBe(true)
    wrapper.instance().toggle()
    wrapper.update()
    expect(wrapper.state('toggled')).toBe(false)
  })
})

describe('render', () => {
  it('should render', () => {
    const snapshot = renderer.create(
      <Accordion>{() => <h1>Hello world</h1>}</Accordion>
    );
    expect(snapshot).toMatchSnapshot()
  })
})