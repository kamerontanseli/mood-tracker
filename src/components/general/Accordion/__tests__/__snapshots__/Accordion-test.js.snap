// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render 1`] = `
<div
  className="Panel"
>
  <h1>
    Hello world
  </h1>
</div>
`;
